packages/legend-extension-dsl-service/src/components/studio/UpdateProjectServiceQuerySetup.tsx

```diff
@@ -58,6 +61,81 @@ import {
 import type { Entity } from '@finos/legend-storage';
 import { extractElementNameFromPath } from '@finos/legend-graph';
 
+const getProjectOptionLabelFormatter = (
+  applicationStore: LegendStudioApplicationStore,
+  setupStore: UpdateProjectServiceQuerySetupStore,
+): ((option: ProjectOption) => React.ReactNode) =>
+  function ProjectOptionLabel(option: ProjectOption): React.ReactNode {
+    const viewProject = (): void => {
```

Comment:
I had to duplicate this code over from`legend-application-studio` because of dependency issue (cannot import `UpdateProjectServiceQuerySetupStore` in `legend-application-studio` ). I see that `UpdateProjectServiceQuerySetupStore`, `WorkspaceSetupStore`, `UpdateServiceQuerySetupStore` share some similar structure and serves similar purpose based on what i understand. Do you think it's better to have a abstract class and have basic stuff over there so that we avoid code duplication?

Reply:
hm, that's odd. I thought `getProjectOptionLabelFormatter` was from `@finos/legend-application-studio` so it would be prefectly fine  for `@finos/legend-extension-dsl-service` to import it right?

Reply:
yeah but i need to add a new input parameter `setupStore` to fetch the `reviewUrl` and display on the `projectOption` so that was causing a dependency issue. Type of `setupStore` would vary with the flow so.